src/providers/php.rs

```diff
@@ -0,0 +1,211 @@
+        let mut cmd = String::new();
+        cmd.push_str("mkdir -p /var/log/nginx && mkdir -p /var/cache/nginx");
+        let nodejs = NodeProvider {};
+        if app.includes_file("composer.json") {
+            cmd.push_str(" && composer install");
+        };
+        if nodejs.detect(app, env)? {
+            if !cmd.is_empty() {
+                cmd.push_str(" && ");
+            }
+            cmd.push_str(
+                nodejs
+                    .install(app, env)?
+                    .as_ref()
+                    .unwrap()
+                    .cmd
+                    .as_ref()
+                    .unwrap()
+                    .as_str(),
+            );
+        }
+        Ok(Some(InstallPhase::new(cmd)))
+    }
+
+    fn build(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::BuildPhase>> {
+        if let Ok(true) = NodeProvider::has_script(app, "prod") {
+            return Ok(Some(BuildPhase::new(
+                NodeProvider::get_package_manager(app) + " run prod",
+            )));
+        }
+        Ok(None)
+    }
+
+    fn start(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::StartPhase>> {
+        Ok(Some(StartPhase::new(format!(
+            "([ -e /app/storage ] && chmod -R ugo+w /app/storage); perl {} {} /nginx.conf && (php-fpm -y {} & nginx -c /nginx.conf)",
+            app.asset_path("transform-config.pl"),
+            app.asset_path("nginx.template.conf"),
+            app.asset_path("php-fpm.conf"),
+        ))))
+    }
+
+    fn static_assets(
+        &self,
+        _app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::app::StaticAssets>> {
+        Ok(Some(static_asset_list! {
+            "nginx.template.conf" => include_str!("php/nginx.template.conf"),
+            "transform-config.pl" => include_str!("php/transform-config.pl"),
+            "php-fpm.conf" => include_str!("php/php-fpm.conf")
+        }))
+    }
+
+    fn environment_variables(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::environment::EnvironmentVariables>> {
+        let mut vars = EnvironmentVariables::new();
+        vars.insert(
+            "PHP_VERSION".to_string(),
+            self.get_php_version(app)
+                .unwrap_or_else(|_| "8.1".to_string()),
+        );
+        if app.includes_file("artisan") {
+            vars.insert("IS_LARAVEL".to_string(), "yes".to_string());
+        }
+        Ok(Some(vars))
+    }
+}
+
+impl PhpProvider {
+    fn get_php_package(&self, app: &crate::nixpacks::app::App) -> anyhow::Result<Option<String>> {
```

Comment:
The PHP versions are duplicated in a few places. I think there should be a single constant map of php version number to nix package and another constant for the default php version. It would make looking up the version and nix package cleaner.